types/react/index.d.ts

```diff
@@ -1201,34 +1201,19 @@ declare namespace React {
      * @see {@link https://react-typescript-cheatsheet.netlify.app/docs/basic/getting-started/forward_and_create_ref/ React TypeScript Cheatsheet}
      * @see {@link forwardRef}
      */
-    interface ForwardRefRenderFunction<T, P = {}> {
-        (props: P, ref: ForwardedRef<T>): ReactNode;
-        /**
-         * Used in debugging messages. You might want to set it
-         * explicitly if you want to display a different name for
-         * debugging purposes.
-         *
-         * Will show `ForwardRef(${Component.displayName || Component.name})`
-         * in devtools by default, but can be given its own specific name.
-         *
-         * @see {@link https://legacy.reactjs.org/docs/react-component.html#displayname Legacy React Docs}
-         */
-        displayName?: string | undefined;
-        /**
-         * defaultProps are not supported on render functions passed to forwardRef.
-         *
-         * @see {@link https://github.com/microsoft/TypeScript/issues/36826 linked GitHub issue} for context
-         * @see {@link https://react.dev/reference/react/Component#static-defaultprops React Docs}
-         */
-        defaultProps?: never | undefined;
-        /**
-         * propTypes are not supported on render functions passed to forwardRef.
-         *
-         * @see {@link https://github.com/microsoft/TypeScript/issues/36826 linked GitHub issue} for context
-         * @see {@link https://react.dev/reference/react/Component#static-proptypes React Docs}
-         */
-        propTypes?: never | undefined;
-    }
+    function ForwardRefRenderFunction<T, P = {}>(props: P, ref: ForwardedRef<T>): ReactNode;
```

Comment:
Only `function` declarations support type inference of higher order functions.
> We only make **higher order function type inferences** when the source and target types are both pure function types, i.e. types with a single call signature and no other members [[1]](https://github.com/microsoft/TypeScript/issues/30650#issuecomment-478258803), [[2]](https://github.com/microsoft/TypeScript/issues/30650#issuecomment-486680485).
https://stackoverflow.com/questions/58469229/react-with-typescript-generics-while-using-react-forwardref/58473012